src/contracts/pods/EigenPodManager.sol

```diff
@@ -173,6 +173,7 @@ contract EigenPodManager is
     ) external onlyDelegationManager {
         require(strategy == beaconChainETHStrategy, InvalidStrategy());
         require(staker != address(0), InputAddressZero());
+        require(int256(shares) >= 0, SharesNegative());
 
```

Comment:
Looks like this function needs to be updated with the TODO removed.
We have above the following check `require(podOwnerDepositShares[podOwner] >= 0, LegacyWithdrawalsNotCompleted());` which should also be added here in the withdrawAsShares function.